README.md

```diff
@@ -1,289 +1,610 @@
+
+An implementation is not compliant if it fails to satisfy one or more of the MUST, MUST NOT, REQUIRED, SHALL, or SHALL NOT requirements for the protocols it implements.  An implementation is compliant if it satisfies all the MUST, MUST NOT, REQUIRED, SHALL, and SHALL NOT requirements for the protocols it implements.
 
-Specification for binding services to runtime applications running in Kubernetes.  
+## Terminology definition
 
-## Conventions
+<dl>
+  <dt>Duck Type</dt>
+  <dd>Any type that meets the contract defined in a specification, without being an instance of a specific concrete type.  For example, for specification that requires a given key on <code>status</code>, any resource that has that key on its <code>status</code> regardless of its <code>kind</code> would be considered a duck type of the specification.</dd>
 
-The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
-"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this
-document are to be interpreted as described in [BCP
-14](https://tools.ietf.org/html/bcp14)
-[[RFC2119](https://tools.ietf.org/html/rfc2119)]
-[[RFC8174](https://tools.ietf.org/html/rfc8174)] when, and only when, they
-appear in all capitals, as shown here.
+  <dt>Service</dt>
+  <dd>Any software that exposes functionality.  Examples include a database, a message broker, an application with REST endpoints, an event stream, an Application Performance Monitor, or a Hardware Security Module.</dd>
 
-## Terminology definition
+  <dt>Application</dt>
+  <dd>Any process, running within a container.  Examples include a Spring Boot application, a NodeJS Express application, or a Ruby Rails application.  <b>Note:</b> This is different than an umbrella application as defined by the Kubernetes SIG, which refers to a set of micro-services.</dd>
+
+  <dt>Service Binding</dt>
+  <dd>The act of or representation of the action of providing information about a Service to an Application</dd>
+
+  <dt>ConfigMap</dt>
+  <dd>A Kubernetes <a href="https://kubernetes.io/docs/concepts/configuration/secret/">ConfigMap</a></dd>
+
+  <dt>Secret</dt>
+  <dd>A Kubernetes <a href="https://kubernetes.io/docs/concepts/configuration/configmap/">Secret</a></dd>
+</dl>
+
+# Provisioned Service
+
+A Provisioned Service resource **MUST** define a `.status.binding.name` which is a `LocalObjectReference` to a `Secret`.  The `Secret` **MUST** be in the same namespace as the resource.  The `Secret` **SHOULD** contain a `type` entry with a value that identifies the abstract classification of the binding.  It is **RECOMMENDED** that the `Secret` also contain a `provider` entry with a value that identifies the provider of the binding.  The `Secret` **MAY** contain any other entry.
+
+## Resource Type Schema
+
+```yaml
+status:
+  binding:
+    name:  # string
+```
+
+## Example Resource
+
+```yaml
+...
+status:
+  ...
+  binding:
+    name: production-db-secret
+```
+
+## Well-known Secret Entries
```

Comment:
There are a variety of built-in "types" with some loosely associated schemas, e.g.
https://github.com/kubernetes/api/blob/afb0842feaf5249db565a4843da5a5de6b28544a/core/v1/types.go#L5585-L5598

Personally, I'm fond of the idea of expecting certain types to correlate with certain schemas (perhaps with additional keys).

Reply:
I think this is right and will happen organically.  If you're a platform for which TLS configuration is important, you'll naturally follow the de facto standards that are emerging without the need for a rigid specification to guide you as those de facto standards evolve.